### ISSUE TITLE
Add target constants for Anchor to use with alternate setTarget method

### ISSUE THREAD

**Issue by joelpop** [vaadin/flow#8346]
Instead of requiring Java users to type one of the four constant target strings, provide them as constants. Example usage would look like

```
    Anchor anchor = new Anchor("https://www.google.com", "Google.");
    anchor.setTarget(AnchorTarget.BLANK);
```

Example implementation might look like:

```
// com.vaadin.flow.component.html.Anchor.java

  /**
   * Sets the target window, tab or frame for this anchor. The target is one of these
   * special values:
   * <ul>
   * <li><code>AnchorTarget.DEFAULT</code>: Removes the target value. This has the same effect as setting the target to <code>AnchorTarget.SELF</code>.
   * <li><code>AnchorTarget.SELF</code>: Opens the link in the current context.
   * <li><code>AnchorTarget.BLANK</code>: Opens the link in a new unnamed context.
   * <li><code>AnchorTarget.PARENT</code>: Opens the link in the parent context, or the
   * current context if there is no parent context.
   * <li><code>AnchorTarget.TOP</code>: Opens the link in the top most grandparent
   * context, or the current context if there is no parent context.
   * </ul>
   *
   * @param target
   *            the target value, not null
   */
  public void setTarget(AnchorTarget target) {
    Objects.requireNonNull(target, "AnchorTarget cannot be null.");
    
    setTarget(target.getValue());
  }
```

and

```
//com.vaadin.flow.component.html.AnchorTarget.java

public enum AnchorTarget {
  /**
   * Remove the target value. This has the same effect as <code>SELF</code>.
   */
  DEFAULT(""),
  /**
   * Open a link in the current context.
   */
  SELF("_self"),
  /**
   * Open a link in a new unnamed context.
   */
  BLANK("_blank"),
  /**
   * Open a link in the parent context, or the current context if there is no
   * parent context.
   */
  PARENT("_parent"),
  /**
   * Open a link in the top most grandparent
   * context, or the current context if there is no parent context.
   */
  TOP("_top");


  private final String value;


  /**
   * @param value the text value to use by an {@code <a>} (anchor) tag.
   */
  private AnchorTarget(String value) {
    this.value = value;
  }


  /**
   * @return
   *         value the text value to use by an {@code <a>} (anchor) tag.
   */
  String getValue() {
    return value;
  }

}
```

**Comment by pleku** [comment]
Probably would be a good idea to add also a new constructor variant which takes the target.

**Comment by jhult** [comment]
I just submitted a pull request: #8397 

A few questions/notes:

1. Are there other valid target values besides the ones @joelpop specified?
	1A. If so, should we deprecate the `setTarget(String)`/`getTarget(String)` methods?
2. Due to other existing tests (I believe they were in `ComponentTest` and `HtmlComponentSmokeTest`), I had to keep the original String methods and add 2 new methods with different names: `setTargetEnum(AnchorTarget)`/`getTargetEnum()`
3. If there is no target (meaning the target was removed since it was set with `""` or `AnchorTarget.DEFAULT`), should `getTargetEnum()` return `AnchorTarget.DEFAULT` or `Optional.empty()`?

**Comment by joelpop** [comment]
> 1. Are there other valid target values besides the ones @joelpop specified?


Yes. The name for a browsing context (a tab, window, or <iframe>) is also valid. See [<a>: The Anchor element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a).

> 1A. If so, should we deprecate the setTarget(String)/getTarget(String) methods?
No

**Comment by joelpop** [comment]
> If there is no target (meaning the target was removed since it was set with "" or AnchorTarget.DEFAULT), should getTargetEnum() return AnchorTarget.DEFAULT or Optional.empty()?

An even harder question to answer is what `getTargetEnum(AnchorTarget)` should return if the target was set with a tab, window, or <iframe> name via `setTarget(String)`.

**Comment by jhult** [comment]
As it relates to 3, `getTargetEnum()` will (currently) return an empty `Optional` if the String value on the `Anchor` is not a recognized `AnchorTarget` value.

**Comment by pleku** [comment]
Closed with #8397
